app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -335,6 +336,24 @@ public void onClick(View v) {
                             .show(getFragmentManager(), TAG);
                 }
                 break;
+            case R.id.detail_controls_download:
+                if (!PermissionHelper.checkStoragePermissions(activity)) {
```

Comment:
Might be a good idea to move this out to a method and do a few null checks. `currentInfo` is async and can be null.